### ISSUE TITLE
My Day

### ISSUE THREAD

**Issue by medensqui** [PlanHubMe/PlanHub#1]
General idea of it:

Students don't get enough sleep.

Students can set ideally when they want to sleep and build a schedule that adapts to their target. They put in after-school activities, their schedule is synced. A hw schedule is suggested and they can adjust hw duration. They mark when they finish their hw. Less time than they thought = More Freetime or making up for the hw time that was earlier subtracted (you'll see in a second) More time than they thought = Subtraction from Freetime > Subtraction from Long-Term > Subtraction from Due Soon > Subtraction from sleep :cry:. This creates a day planner (or weekly view) that allows students to get as much sleep as they can without sacrificing schoolwork.

**Comment by thatoddmailbox** [comment]
@medensqui We need a name for this. (a short one, to put on the tab). "Schedule" wouldn't work unless we rename the current schedule tab...

**Comment by medensqui** [comment]
@thatoddmailbox What about My Day? Stealing a bit from Whipplehill here but still.

**Comment by thatoddmailbox** [comment]
@medensqui That's a good idea!

**Comment by c20et** [comment]
Hello

**Comment by medensqui** [comment]
@c20et Hi Eeshan!

**Comment by thatoddmailbox** [comment]
@c20et Hello

**Comment by medensqui** [comment]
@thatoddmailbox So for the interface

This may require a little bit of algorithms...

I was thinking a drag and drop

You would put in your sleep target i.e. 11-7

It would sync your schedule
7-8:10 = Getting ready/going to school
8:10-3:15 = Schedule sync + Labby sync for labs 
During school day you can put hw in for labs
This will subtract time for hw if it was being worked on during afternoon
After 3:15 Student can put in After School Commitments (this is where commutes can go)
They can click a button to suggest (algorithm)
Prioritize due next day (default length can be set in preferences, default default would be 45?)
Then due soon
Then long term
Then put in free time
Or can drag and drop school work from a side bar
Any non-used time = Freetime

Algorithm for less/more time

When a student marks an assignment as done, check time student was scheduled to start vs. time he marked done

If less time:
Add back lost time in more time subtraction
Then add freetime
If more time:
Subtract from freetime
Subtract from long term (evenly as you can)
Subtract from due soon (evenly as you can)
Subtract from sleep

Sorry for length.

On a scale of 1-10 how hard is this in code?

**Comment by thatoddmailbox** [comment]
@medensqui 
- You should be able to change the getting ready part, if going to school takes longer (if you live far away, commute, etc)
- What do you mean by suggest after school commitments? Like it would have a list of clubs? We could ask clubs for their events and automagically put them in! Or maybe someone in student gov. has a list of events
- np about length

The scale depends if you're talking about the pretty frontend or the backend.

Frontend probably like 4-5, because the dragging might be difficult (though you could try some jQuery UI stuff)

Backend probably like 5-6 (excluding easy schedule sync or making Labby) because the algorithms (telling it how to prioritize stuff) might be tricky. Though I'm not actually that sure if it will be that hard...

**Comment by medensqui** [comment]
@thatoddmailbox

Anytime from when you wake up to 8:10 is getting ready/ commute

Cool. 

Users would input their own commitments, however it would also have an option for repeating commitments 

Awesome!

Sent from FirstClass with my iPhone

**Comment by thatoddmailbox** [comment]
@medensqui I think if we could get clubs to publish info that would be nice...

but it is possible! This actually might just supersede the "Schedule" feature (as it's literally what I thought about but plus the smart prioritization stuff)...

**Comment by medensqui** [comment]
@thatoddmailbox Also, Studer, what if we had like a "check all clubs you participate in" option at the beginning and it would suggest club events. This could also work with the lab management part of My Day also

**Comment by thatoddmailbox** [comment]
@medensqui Exactly. We ask clubs "please send us events" or maybe "please confirm this list from the planner". Then, when you click on the afterschool commitments thingy, it will have a button to select clubs.

**Comment by medensqui** [comment]
@thatoddmailbox 
1. How do I make tags?
2. Do I need to know JQuery for My Day? Like for the drag and drop?

**Comment by thatoddmailbox** [comment]
@medensqui 
1. Go to the issues page, scroll to the top, click "Labels".
2. There is a lot of jQuery used not just for My Day but in general. HOWEVER, excepting the stuff that will happen with My Day, it's not hard or complicated. You could just leave the drag-and-drop part alone.

**Comment by medensqui** [comment]
@thatoddmailbox Ok

Codeacademy? Is that how I should learn?

**Comment by medensqui** [comment]
@thatoddmailbox IT MUST BE DRAGGY AND DROPPY

Or I guess it could be click with a pretty clicky animation

**Comment by thatoddmailbox** [comment]
@medensqui Probably. Also when I said HTML I meant Jade. (Jade is a fancy way of writing HTML)

Example:

```
div
    h1 Something
```

turns in to

```
<div><h1>Something</h1></div>
```

**Comment by thatoddmailbox** [comment]
@medensqui we can keep it "DRAGGY AND DROPPY"! but what do you mean by a pretty clicky animation?

**Comment by medensqui** [comment]
@thatoddmailbox OH YES PLS

OK I WANT IT DRAGGY AND DROPPY

**Comment by thatoddmailbox** [comment]
@medensqui YAY

**Comment by medensqui** [comment]
@thatoddmailbox ALSO HOW THE HECK (for barkoff's sake) DO I LEARN JQUERY, JS, CSS, AND ALL THAT STUFF

**Comment by thatoddmailbox** [comment]
@medensqui 
Codecademy?

CSS is not that complicated (you maybe could get away without knowing it?).
JS and jQuery is on Codecademy, and if you know HTML you basically know Jade.

**Comment by medensqui** [comment]
@thatoddmailbox True. Also CSS is part of the HTML track that I can do

Ok I'll study JS and jQuery.

And C+ is pretty much CSS w/ a little bit of JS right?

**Comment by thatoddmailbox** [comment]
@medensqui Yes. C+ uses _some_ HTML for the options page, but that's about it.

**Comment by medensqui** [comment]
@thatoddmailbox Ok thanks!
So HTML/CSS, JS, jQuery?

**Comment by thatoddmailbox** [comment]
@medensqui Pretty much. Again, PlanHub uses Jade, but Jade is just a fancy HTML, so.

**Comment by thatoddmailbox** [comment]
@medensqui Should we just roll this into Schedule? Replace the tab with "My Day"

**Comment by medensqui** [comment]
@thatoddmailbox Yeah I think so. Schedule is just slightly better schedules.dalton.org (Labby integration). But My Day replaces this and more

**Comment by thatoddmailbox** [comment]
@medensqui Exactly. My Day will do all the schedule stuff anyways, and it would be confusing to have two tabs that do essentially the same thing.

**Comment by medensqui** [comment]
@thatoddmailbox So does that mean you're starting work on it? :smile: 
Tell me when you have a very rough Alpha 1.0 so that way I can see if it looks like what I thought it would

**Comment by thatoddmailbox** [comment]
@medensqui Not yet, but I will.

Do you think we should release My Day on the first day, _or_ should we have a beta-ish version of My Day only available to certain testers while the rest is for everyone else.

**Comment by medensqui** [comment]
@thatoddmailbox A bit of both. Day 1 have in big letter "WORK IN PROGRESS" and have screenshots and a fancy tourish thing but don't let them use it. Also we should have a release date for it by then

**Comment by thatoddmailbox** [comment]
@medensqui I guess by Thursday or Friday? If we make the club + announcement on Thursday we should have it publicly available.

**Comment by medensqui** [comment]
@thatoddmailbox True. Also, I meant release date for My Day, but Thursday or Friday works too. Lechich still hasn't replied about club requirements. Can you ask Neiers about club in house? Idk if he will want to though.

**Comment by medensqui** [comment]
@thatoddmailbox And will he count as a faculty advisor?

**Comment by thatoddmailbox** [comment]
@medensqui Idk. I think we should finish the charter first and _maybe_ have the website a bit more stable so we have something to show him.

**Comment by thatoddmailbox** [comment]
@medensqui I set it to Thursday for now.

**Comment by medensqui** [comment]
@thatoddmailbox Ok. 
DO NOT SUBMIT THE CHARTER. That's the one for returning clubs. I'm trying to get the one for new clubs

**Comment by thatoddmailbox** [comment]
@medensqui Ik. I mean we should have the gdoc readyish.

**Comment by medensqui** [comment]
@thatoddmailbox Ah yeah true

**Comment by c20et** [comment]
Ok so it should force the user to sit down and fully accomplish the task inputed from plannerHub. There should be also a countdown timer for the free time you have. So if the user inputs that they sleep at 11 it should show how much time they actually have. This would motivate the user.

**Comment by thatoddmailbox** [comment]
@c20et we can't really force the user to do that - we don't have control over the whole computer, it's just a webpage. Also idk if the countdown would work because if you get the estimates wrong it's somewhat depressing when it says "oh you have free time now" when you don't.

**Comment by medensqui** [comment]
@c20et  Studer is right. But good idea though.

@thatoddmailbox Wait but couldn't we estimate it based on how much freetime you have currently on your schedule

**Comment by thatoddmailbox** [comment]
@medensqui I mean if you mess up the estimates for HW and then the program miscalculates the free time

**Comment by medensqui** [comment]
@thatoddmailbox Ah true. So no timer. Or we could .put something like that in the overview

**Comment by thatoddmailbox** [comment]
@medensqui We could put a timer, that would be fine. Just idk if it will be so accurate.

**Comment by medensqui** [comment]
@thatoddmailbox ok and if people complain we remove it

**Comment by thatoddmailbox** [comment]
@medensqui or add a button to hide it

**Comment by medensqui** [comment]
@thatoddmailbox Good idea. READ MY POST ON CUSTOM

**Comment by thatoddmailbox** [comment]
@medensqui I DID GEEZ

**Comment by thatoddmailbox** [comment]
Moving this out of Public release because it seems like this will not make it...

**Comment by medensqui** [comment]
@thatoddmailbox Yeah...
It's really important though, in history class w/ people complaining about how little sleep they get

**Comment by willbarkoff** [comment]
@medensqui Earlier how you were complaining about knowing CSS it is really easy to learn. I learned it in like 3 hrs.

**Comment by willbarkoff** [comment]
@medensqui and JS I get away without knowing because I know ruby and ruby can do some things like JS and I sort of know JS.

**Comment by willweinstein** [comment]
@nywillb REALLY? I GET THROUGH CODING BY WATCHING STUDER UNTIL I GET THE SLIGHTEST UNDERSTANDING THEN WRITE BAD CODE AND PRETEND MY LACK OF KNOWLEDGE WAS JUST MISTAKES FROM RUSHING

**Comment by medensqui** [comment]
OK HERES WHAT WE FKIN NEED
- [ ] Make the close button thingy werk
- [ ] New event buttn
- [ ] Backend algorithms (SEE SLACK)
- [ ] Make tag defaults
- [ ] Scrape Schedule and put in

**Comment by thatoddmailbox** [comment]
@medensqui 

> Homework should disappear from list

The thing with this part is if you have homework you want to split over multiple days. (I'm not going to study for a test all in one day, I will do some amount every day). Maybe if you drag the event smaller, it will reappear in the right-hand pane and you can drag a new event to finish it. (not sure if that made sense)

**Comment by medensqui** [comment]
True!

**Comment by medensqui** [comment]
@thatoddmailbox
